### PR TITLE
Add tests for hd keys and code improvements 

### PR DESCRIPTION
**Pull request by 0xmaayan**
### Description
following https://github.com/aptos-labs/aptos-ts-sdk/pull/146#pullrequestreview-1707312446
1. Remove KeyType enum
2. Add hd tests
3. Making @alinush happy

### Test Plan
pnpm test

### CODE REVIEW

File: src/core/crypto/ed25519.ts [hunk 1]
```diff
@@ -99,6 +99,12 @@ export class Ed25519PrivateKey extends PrivateKey {
    */
   static readonly LENGTH: number = 32;
 
+  /**
+   * The Ed25519 key seed to use for BIP-32 compatibility
+   * See more {@link https://github.com/satoshilabs/slips/blob/master/slip-0010.md}
+   */
+  static readonly SEED = "ed25519 seed";
```
Comment (0xmaayan) [inline src/core/crypto/ed25519.ts:106]: not sure about this name

Comment (gregnazario) [reply src/core/crypto/ed25519.ts:106]: Maybe call it... BIP32_SEED

File: src/core/account.ts [hunk 2]
```diff
@@ -220,13 +220,13 @@ export class Account {
...
-    let privateKey;
+    let privateKey: PrivateKey;
     switch (scheme) {
       case SigningSchemeInput.Secp256k1Ecdsa:
-        privateKey = new Secp256k1PrivateKey(Secp256k1PrivateKey.fromDerivationPath(path, mnemonic));
+        privateKey = Secp256k1PrivateKey.fromDerivationPath(path, mnemonic);
         break;
       case SigningSchemeInput.Ed25519:
-        privateKey = new Ed25519PrivateKey(Ed25519PrivateKey.fromDerivationPath(path, mnemonic));
+        privateKey = Ed25519PrivateKey.fromDerivationPath(path, mnemonic);
```
Comment (gregnazario) [inline src/core/account.ts:229]: much nicer

File: tests/unit/hdKey.test.ts [hunk 3]
```diff
@@ -61,4 +61,106 @@ describe("Hierarchical Deterministic Key (hdkey)", () => {
...
+          },
+        ],
+      },
+    ];
+
+    ed25519.forEach(({ seed, vectors }) => {
+      vectors.forEach(({ chain, private: privateKey }) => {
+        it(`should generate correct key pair for ${chain}`, () => {
+          // eslint-disable-next-line @typescript-eslint/dot-notation
+          const key = Ed25519PrivateKey["fromDerivationPathInner"](chain, seed.toUint8Array());
```
Comment (gregnazario) [inline tests/unit/hdKey.test.ts:109]: wondering why not dot?  Is that cause the private function?

Swear you're a magician at typescript :)

**Review by gregnazario — APPROVED** [review]

File: src/core/crypto/ed25519.ts (hunk 1 above)
Comment (alinush) [reply src/core/crypto/ed25519.ts:106]: This is a SLIP-0010 seed, not BIP-32.

**Review by alinush — APPROVED** [review]

File: tests/unit/hdKey.test.ts (hunk 3 above)
Comment (0xmaayan) [reply tests/unit/hdKey.test.ts:109]: yep! private static method

**Pull request merged**